atlasdb-api/src/main/java/com/palantir/atlasdb/config/LeaderConfig.java

```diff
@@ -85,6 +85,8 @@ protected final void check() {
 
     @Value.Derived
     public LockLeader whoIsTheLockLeader() {
-        return LockLeader.iAmTheLockLeader(lockCreator().equals(localServer()));
```

Comment:
The problem with the inlined version is, the logic of what the enum values mean is encoded in LeaderConfig, rather than in LockLeader. I'd vote for `LockLeader.fromBoolean`.